### ISSUE TITLE
VB -> C#: If interface and class member differ only by capitalization, the class member's case is changed.

### ISSUE THREAD

**Issue by HeinziAT** [icsharpcode/CodeConverter#963]
### VB.Net input code
```vbnet
Public Interface I
    Public Sub A()
End Interface

Public Class C
    Implements I

    Public Sub a() Implements I.A
    End Sub
End Class
```

### Erroneous output
```C#
public partial interface I
{
    public void A();
}

public partial class C : I
{

    public void A()
    {
    }
}
```

### Expected output
```C#
public partial interface I
{
    public void A();
}

public partial class C : I
{

    public void a()
    {
    }

    void I.A() => a();
}
```

OR
```C#
public partial interface I
{
    public void A();
}

public partial class C : I
{
    /* TODO WARNING: Case changed from a to A (to match supertype) */
    public void A()
    {
    }
}
```

### Details
* Product in use: [icsharpcode.github.io/CodeConverter](https://icsharpcode.github.io/CodeConverter/)
* Version in use: 9.0.4.0
* Why is this a problem?
    * VB.NET is case-insensitive, but CIL isn't. Thus, if you convert a library from VB to C# and the capitalization of a public member changes, you break binary-compatibility, i.e., an application compiled to the VB version of your library won't work with the C# version.
    * You also break source-compatibility, if your VB library is used by C# applications.
* The converter already produces the expected output (method + second method with explicit interface implementation) if you replace `Public Sub a() Implements I.A` by `Public Sub b() Implements I.A`. So the fix might be as easy as changing a case-insensitive comparison to a case-sensitive one.
* I admit that `a`/`A` is a contrived example, but a real-life use case would be an abbreviation that is capitalized differently. In our code base, we stumbled upon this issue with `Sub AddSubSQL` (interface) vs `Sub AddSubSql` (class).
* Note that this change will trigger a warning: *CS3005 Identifier 'C.a' differing only in case is not CLS-compliant*, so it might make sense to additionally add the a `[CLSCompliant(false)]` attribute to the method. Unfortunately, I don't know enough about CLS compliance to be able to evaluate the impact of this change.

**Comment by GrahamTheCoder** [comment]
I think the code here tries to solve a similar problem for namespaces declared with multiple casings.
You can see there how it was pretty awkward in that case because the code analysis library loses the information so manual syntax analysis was needed. It may be similar for this case.
https://github.com/icsharpcode/CodeConverter/blob/8ccc9ce43c3f803de7e6dabff7d23c8d98522d35/CodeConverter/CSharp/CommonConversions.cs#L206-L222

I'd never imagined that someone would intentionally vary their casing between methods on interfaces and methods on classes, but the ultimate goal *is* to keep binary compatibility where possible. I'd say this is reasonably low priority since I don't expect this to happen often or systematically and it's probably reasonably easy to fix

**Comment by HeinziAT** [comment]
I agree, it's a rare edge case which should be low priority, especially if it's not an easy fix.

In our case, the varied casing was unintentional: One class used "...Sql" for the method name suffix, the other class used "...SQL", without anyone noticing. Years later, we decided that it would be nice to have a common interface for both classes. By then, it was too late the change the casing in the classes.
